signac/core/dict_manager.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2019 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"Basic wrapper to access multiple different data stores."
+import os
+import re
+import errno
+import uuid
+
+from ..common import six
+
+
+class DictManager(object):
+
+    cls = None
+    suffix = None
+
+    __slots__ = ['_prefix', '_dict_registry']
+
+    def __init__(self, prefix):
+        assert self.cls is not None
+        assert self.suffix is not None
```

Comment:
In the interest of supporting future developers who find alternative uses for subclassing DictManager I think it's worth adding explicit error messages here indicating that the subclasses need to define both of these class attributes.